src/modules/roc_fec/reader.cpp

```diff
@@ -76,32 +73,55 @@ packet::PacketPtr Reader::read_() {
     fetch_packets_();
 
     if (!started_) {
-        packet::PacketPtr pp = source_queue_.head();
-        if (pp) {
-            if (!has_source_) {
-                source_ = pp->rtp()->source;
-                has_source_ = true;
-            }
-            cur_sbn_ = pp->fec()->source_block_number;
-            drop_repair_packets_from_prev_blocks_();
-        }
+        packet::PacketPtr pp = read_first_();
 
         if (!pp || pp->fec()->encoding_symbol_id > 0) {
             return source_queue_.read();
```

Comment:
Do we still need this condition?

```
pp->fec()->encoding_symbol_id > 0
```

It probably made sense only when we relied on the marker bit? Not sure but anyway it looks unnecessary.

If we don't need it anymore, this check:

```
        if (!pp || pp->fec()->encoding_symbol_id > 0) {
            return source_queue_.read();
        }
```

could be reduced to:

```
        if (!pp) {
            return NULL;
        }
```

and we'll get the ability to repair lost packets event in the very fist block.

Reply:
Hm.. Why we can simply return NULL instead of `return source_queue_.read()`?
We have an `OR` condition in the mentioned if block.

Reply:
Because there is no difference between this:

```
        packet::PacketPtr pp = source_queue_.head();

        if (!pp) {
            return source_queue_.read();
        }
```

and this:

```
        packet::PacketPtr pp = source_queue_.head();

        if (!pp) {
            return NULL;
        }
```

(well, there is no *practical* difference)

Reply:
Agree. Lets extract it into separate PR? I also want to cover this with test.

Reply:
Sure.

Reply:
#186